### PR TITLE
Social tlo added

### PR DESCRIPTION
**Pull request by DerpleMQ2**
- Add new Social TLO to access social buttons.
Supports the following Members:
 . Name
 . Cmd(0-4)
 . TimerBegin
 . TimerDuration
 . Color

### CODE REVIEW

File: src/main/datatypes/MQSocialType.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	ScopedTypeMember(SocialTypeMembers, Color);
+}
+
+bool MQSocialType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest)
+{
+	if (!pLocalPC || !pLocalPlayer)
+		return false;
+
+	if (!pCursorAttachment)
+		return false;
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:51]: This doesn't feel like it needs to be here.

File: src/main/datatypes/MQSocialType.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		Dest.Type = mq::datatypes::pStringType;
+		Dest.Ptr = &DataTypeTemp[0];
+		return true;
+
+	case SocialTypeMembers::Cmd:
+		if (Index[0])
+		{
+			if (IsNumber(Index))
+			{
+				int idx = GetIntFromString(Index, -1);
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:75]: GetIntFromString adequately covers the first two conditions well enough, I don't think you need them.

File: src/main/datatypes/MQSocialType.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+bool MQSocialType::dataSocial(const char* szIndex, MQTypeVar& Ret)
+{
+	Ret.Ptr = nullptr;
+	Ret.Type = pSocialType;
+
+	if (IsNumber(szIndex))
+	{
+		Ret.Int = GetIntFromString(szIndex, -1);
+	}
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:128]: Likewise, don't need the IsNumber check.
You want to be setting Ret.Int anyways, or you'll get an error in GetMember when you try to read it.

File: src/main/datatypes/MQSocialType.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	{
+		return pSocialList[socialIdx].Line[lineIdx];
+	}
+
+	return "";
+}
+
+bool MQSocialType::dataSocial(const char* szIndex, MQTypeVar& Ret)
+{
+	Ret.Ptr = nullptr;
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:122]: Don't need this line.

File: src/main/datatypes/MQSocialType.cpp [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "pch.h"
+
+#include "MQ2DataTypes.h"
+#include "MQ2DataTypes.h"
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:18]: extra include

File: src/main/datatypes/MQ2DataTypes.h [hunk 6]
```diff
@@ -1495,6 +1495,22 @@ class MQCursorAttachmentType : public MQ2Type
...
+{
+public:
+	MQSocialType();
+
+	bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override;
+
+	static bool dataSocial(const char* szIndex, MQTypeVar& Ret);
+
+private:
+	const char* getCommandLine(const int socialIdx, const int lineIdx);
```
Comment (brainiac) [inline src/main/datatypes/MQ2DataTypes.h:1511]: this could be a static function in the cpp

File: src/main/datatypes/MQSocialType.cpp (hunk 1 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:51]: 👀

Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:51]: Fixed.

File: src/main/datatypes/MQSocialType.cpp (hunk 2 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:75]: Fixed.

File: src/main/datatypes/MQSocialType.cpp (hunk 3 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:128]: Fixed.

File: src/main/datatypes/MQSocialType.cpp (hunk 4 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:122]: Fixed.

File: src/main/datatypes/MQSocialType.cpp (hunk 5 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp:18]: Fixed.

https://github.com/macroquest/macroquest/blob/0dd4e0a65f61856079f4457923a42e9a209d5460/src/main/datatypes/MQCursorAttachmentType.cpp#L17

File: src/main/datatypes/MQ2DataTypes.h (hunk 6 above)
Comment (DerpleMQ2) [reply src/main/datatypes/MQ2DataTypes.h]: done

File: src/main/datatypes/MQSocialType.cpp [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+#include "pch.h"
+
+#include "MQ2DataTypes.h"
+
+namespace mq::datatypes {
+
+//============================================================================
+
+#define MAX_SOCIAL 120
+#define MAX_CMD_LINES 5
```
Comment (brainiac) [inline src/main/datatypes/MQSocialType.cpp:24]: Oh, i missed these on my first review. we have SOCIAL_NUM_LINES and NUM_SOCIAL_PAGES, SOCIALS_PER_PAGE

Comment (DerpleMQ2) [reply src/main/datatypes/MQSocialType.cpp]: fixed.

**Pull request merged**